unpack.c

```diff
@@ -5,52 +5,79 @@
 #include <archive_entry.h>
 #include <emscripten.h>
 
+typedef struct {
+    char* filename;
+    uint8_t* data;
+    size_t data_size;
+} FileData;
+
 EMSCRIPTEN_KEEPALIVE
-char* extract_archive(uint8_t* inputData, size_t inputSize, size_t* outputSize) {
+FileData* extract_archive(uint8_t* inputData, size_t inputSize, size_t* outputSize, size_t* fileCount) {
     struct archive* archive;
     struct archive_entry* entry;
-    char* outputBuffer = NULL;
-    size_t bufferCapacity = 0;
-    size_t totalBytes = 0;
+    FileData* files = NULL;
+    size_t files_count = 0;
 
     archive = archive_read_new();
-    archive_read_support_filter_all(archive); 
-    archive_read_support_format_all(archive); 
-   
+    archive_read_support_filter_all(archive);
+    archive_read_support_format_all(archive);
+
     if (archive_read_open_memory(archive, inputData, inputSize) != ARCHIVE_OK) {
         fprintf(stderr, "Error opening archive: %s\n", archive_error_string(archive));
         archive_read_free(archive);
         return NULL;
     }
+    printf("Archive opened successfully.\n");
 
     while (archive_read_next_header(archive, &entry) == ARCHIVE_OK) {
         const char* filename = archive_entry_pathname(entry);
         size_t entrySize = archive_entry_size(entry);
+        printf("Extracting file: %s, size: %zu\n", filename, entrySize);
+
+        files = realloc(files, sizeof(FileData) * (files_count + 1));
+        if (!files) {
+            fprintf(stderr, "Memory allocation error for FileData array.\n");
+            archive_read_free(archive);
+            return NULL;
+        }
 
-        if (totalBytes + entrySize > bufferCapacity) {
-            bufferCapacity = totalBytes + entrySize + 1024;
-            outputBuffer = realloc(outputBuffer, bufferCapacity);
+        files[files_count].filename = strdup(filename);
+        files[files_count].data = malloc(entrySize);
+        printf("Setting data_size for file: %s, size: %zu\n", filename, entrySize);
```

Comment:
Does these get printed in the JS console? We may want to get rid of those logs once we have a stable library.

Reply:
Hello @martinRenou yes, all C printing are in js console. I suggest to keep error logging

Reply:
I think a nicer approach would be for the C function to return an `int`:
- 0 if there was no error
- 1 otherwise (you could also use different integer values for different errors)

Then on the TypeScript side we check that value and throw a proper exception in the JS world.

Reply:
Let's not make this a blocker for this PR though! I will open an issue to track this